src/ansible_builder/_target_scripts/introspect.py

```diff
@@ -203,11 +216,24 @@ def simple_combine(reqs):
                 continue
 
             base_line = line.split('#')[0].strip()
+            name_match = REQ_NAME_RE.match(base_line)
+            name = REQ_NORM_RE.sub('-', name_match.group(1))
+            if name in exclude and collection not in {'user', 'exclude'}:
+                logger.debug('# Explicitly excluding requirement %s from %s', name, collection)
+                continue
+            if name in EXCLUDE_REQUIREMENTS and collection not in {'user', 'exclude'}:
+                logger.debug('# Excluding requirement %s from %s', name, collection)
+                continue
+
             if base_line in consolidated:
                 i = consolidated.index(base_line)
-                fancy_lines[i] += f', {collection}'
+                if not name_only:
+                    fancy_lines[i] += f', {collection}'
```

Comment:
We need to kill off fancy lines lines, and just spit back out a concatenation, but not consolidate duplicates.